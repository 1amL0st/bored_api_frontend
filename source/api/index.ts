import { useSelector } from 'react-redux';

import {
  ActionAddSavedActivities,
  ActionRemoveSavedActivities,
  ActionClearSavedActivities,
  ActionAddOfferActivity,
  ActionClearOfferActivities,
  ActionRemoveOfferActivities,
} from 'store/actions/Activities';

import { ActionShouldShowWelcomeWindow } from 'store/actions/Session';

import { AppStore } from 'store';
import { IAppStore } from 'store/reducers/RootReducer';

export const ActivityTypes = [
  'education',
  'recreational',
  'social',
  'diy',
  'charity',
  'cooking',
  'relaxation',
  'music',
  'busywork',
];

export interface IActivity {
  name: string;
  type: string;
  accessibility: number;
  participants: number;
  price: number;
  addDate: Date;
}

export interface ServerResponse {
  activity: string;
  type: string;
  participants: number;
  price: number;
  link: string;
  key: string;
  accessibility: number;
}

export const API = {
  castServerResponseToActivity(response: ServerResponse): IActivity {
    return {
      name: response.activity,
      accessibility: response.accessibility,
      participants: response.participants,
      type: response.type,
      price: response.price,
      addDate: new Date(),
    };
  },

  fetchActivity() {
    const url = 'https://www.boredapi.com/api/activity/';
    fetch(url).then((response) => {
      response.json().then((json: ServerResponse) => {
        AppStore.dispatch(
          ActionAddOfferActivity([API.castServerResponseToActivity(json)])
        );
      });
    });
  },

  clearSavedActivities() {
    AppStore.dispatch(ActionClearSavedActivities());
  },

  useSavedActivities(): Array<IActivity> {
    return useSelector((store: IAppStore) => {
      return store.Activities.saved;
    });
  },

  useOfferActivities(): Array<IActivity> {
    return useSelector((store: IAppStore) => {
      return store.Activities.offers;
    });
  },

  clearOfferActivities() {
    AppStore.dispatch(ActionClearOfferActivities());
  },

  setShouldShowWelcomeWindow(shouldShow: boolean) {
    AppStore.dispatch(ActionShouldShowWelcomeWindow(shouldShow));
  },

  getShouldShowWelcomeWindow(): boolean {
    return AppStore.getState().Session.shouldShowWelcomeWindow;
  },

  removeOfferActivities(activities: Array<IActivity>) {
    AppStore.dispatch(ActionRemoveOfferActivities(activities));
  },

  removeSavedActivities(activities: Array<IActivity>) {
    AppStore.dispatch(ActionRemoveSavedActivities(activities));
  },

  addSavedActivities(activities: Array<IActivity>) {
    AppStore.dispatch(ActionAddSavedActivities(activities));
  }
};
