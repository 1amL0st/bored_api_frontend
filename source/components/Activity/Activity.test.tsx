import React from 'react';
import renderer from 'react-test-renderer';

import { Activity } from './index';
import { IActivity } from 'api';

describe('My Test', () => {
  test('Activity test', () => {
    const activity: IActivity = {
      name: 'Learn somethin new',
      type: 'education',
      accessibility: 30,
      participants: 20,
      price: 10,
      addDate: new Date(),
    };
    const component = renderer.create(
      <Activity isSelected={false} activity={activity} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
