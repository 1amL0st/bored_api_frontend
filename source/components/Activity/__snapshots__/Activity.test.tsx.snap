// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Test Activity test 1`] = `
<div
  className="activity"
  onClick={[Function]}
>
  <span
    title="Activity name"
  >
    Learn somethin new
  </span>
  <span
    className="activity-type"
    title="Type: education"
  >
    <div
      className="activity-type-indicator"
      style={
        Object {
          "backgroundColor": "#d8221c",
        }
      }
    />
  </span>
  <span
    title="The number of people that this activity could involve"
  >
    20
  </span>
  <span
    title="A factor describing how possible an event is to do with zero being the most accessible"
  >
    30
  </span>
  <span
    title="A factor describing the cost of the event with zero being free"
  >
    10
  </span>
</div>
`;
